tools/bridge/bridge/event_fetcher.py

```diff
@@ -110,5 +104,8 @@ def fetch_events(self, poll_interval: int) -> None:
         self.logger.debug("Start event fetcher.")
 
         while True:
-            self.fetch_events_not_seen()
-            sleep(poll_interval)
+            events = self.fetch_some_events()
+            for event in events:
+                self.event_queue.put(event)
+            if not events:
```

Comment:
If I understood correctly, one of the goals of the PR was to put the artificial events on the queue if we didn't get any real ones. But it doesn't look like this is actually happening, or am I missing something?

Reply:
No, you're not missing anything. This isn't implemented yet, but we had a bug in the old code, where it called sleep, when there were already new blocks. This has been fixed.